### PR TITLE
Fix for failing compiler server test SimpleMSBuild, that uses Roslyn msb...

### PR DESCRIPTION
**Pull request by mavasani**
...uild tasks.

Add "ErrorLog" and "ReportAnalyzer" properties to our build tasks for the corresponding properties added to Microsoft.CSharp.CurrentVersion.targets and Microsoft.VisualBasic.CurrentVersion.targets.

@agocke @tmeschter @heejaechang @Srivatsn @jasonmalinowski can you please review?

### CODE REVIEW

File: src/Compilers/Core/MSBuildTask/ManagedCompiler.cs [hunk 1]
```diff
@@ -535,7 +547,10 @@ protected internal virtual void AddResponseFileCommands(CommandLineBuilderExtens
             commandLine.AppendPlusOrMinusSwitch("/optimize", this._store, "Optimize");
             commandLine.AppendSwitchIfNotNull("/out:", this.OutputAssembly);
             commandLine.AppendSwitchIfNotNull("/ruleset:", this.CodeAnalysisRuleSet);
+            commandLine.AppendSwitchIfNotNull("/errorlog:", this.ErrorLog);
             commandLine.AppendSwitchIfNotNull("/subsystemversion:", this.SubsystemVersion);
+            // TODO: uncomment the below line once "/reportanalyzer" switch is added to compiler.
```
Comment (mavasani) [inline src/Compilers/Core/MSBuildTask/ManagedCompiler.cs]: @heejaechang you will need to uncomment this once this switch is added to compiler.

File: src/Compilers/Core/MSBuildTask/ManagedCompiler.cs [hunk 2]
```diff
@@ -89,6 +89,12 @@ public bool EmitDebugInformation
             get { return _store.GetOrDefault("EmitDebugInformation", false); }
         }
 
+        public string ErrorLog
+        {
+            set { _store["ErrorLog"] = value; }
```
Comment (agocke) [inline src/Compilers/Core/MSBuildTask/ManagedCompiler.cs]: Please use nameof for a new argument.

**Comment by agocke** [comment]
I don't see the changes to the CurrentVersion targets reflect in the Core targets. Was that in a separate change?

**Comment by heejaechang** [comment]
:+1:

**Comment by mavasani** [comment]
@agocke I didn't get you, the "CoreCompile" target is only defined in language specific targets files, what changes were you expecting core targets?

**Comment by agocke** [comment]
@mavasani These files: https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets

https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/MSBuildTask/Microsoft.VisualBasic.Core.targets

**Comment by mavasani** [comment]
@agocke Ah, thanks! I didn't know about those Roslyn specific ones.

I have made the changes and updated the PR.

**Comment by agocke** [comment]
LGTM

**Pull request merged**